src/overlays/actors/ovl_En_Baba/z_en_baba.c

```diff
@@ -128,142 +149,145 @@ static TrackOptionsSet sTrackOptions = {
+            this->flags |= ENBABA_END_CONVERSATION;
             break;
 
         case 0x65A:
-            Actor_ChangeFocus(&this->actor, play, &this->unk_144->actor);
-            this->unk_1E0 = 0x65B;
+            Actor_ChangeFocus(&this->actor, play, &this->bombShopkeeper->actor);
+            this->textId = 0x65B;
             break;
 
         case 0x65B:
-            Actor_ChangeFocus(&this->unk_144->actor, play, &this->actor);
-            this->unk_1E0 = 0x65C;
+            Actor_ChangeFocus(&this->bombShopkeeper->actor, play, &this->actor);
+            this->textId = 0x65C;
             break;
 
         case 0x65C:
-            Actor_ChangeFocus(&this->actor, play, &this->unk_144->actor);
-            this->unk_1E0 = 0x65D;
+            Actor_ChangeFocus(&this->actor, play, &this->bombShopkeeper->actor);
+            this->textId = 0x65D;
             gSaveContext.save.weekEventReg[73] |= 2;
-            this->unk_40A |= 1;
+            this->flags |= ENBABA_END_CONVERSATION;
             break;
 
         case 0x65E:
-            Actor_ChangeFocus(&this->actor, play, &this->unk_144->actor);
-            this->unk_1E0 = 0x65F;
-            this->unk_40A |= 1;
+            Actor_ChangeFocus(&this->actor, play, &this->bombShopkeeper->actor);
+            this->textId = 0x65F;
+            this->flags |= ENBABA_END_CONVERSATION;
             break;
 
         case 0x2A34:
             if (INV_CONTENT(ITEM_MASK_BLAST) == ITEM_MASK_BLAST) {
-                this->unk_40A |= 1;
-                this->unk_1E0 = 0x2A36;
+                this->flags |= ENBABA_END_CONVERSATION;
+                this->textId = 0x2A36;
                 break;
             }
-            this->unk_40A |= 0x20;
-            this->unk_1E0 = 0x2A35;
+            this->flags |= ENBABA_GIVE_BLAST_MASK;
+            this->textId = 0x2A35;
             break;
 
         case 0x2A35:
-            this->unk_40A |= 1;
-            this->unk_1E0 = 0x2A36;
+            this->flags |= ENBABA_END_CONVERSATION;
+            this->textId = 0x2A36;
             break;
 
         case 0x2A30:
         case 0x2A31:
-            this->unk_40A |= 1;
+            this->flags |= ENBABA_END_CONVERSATION;
             break;
     }
 
-    Message_StartTextbox(play, this->unk_1E0, &this->actor);
-    if (this->unk_40A & 1) {
-        if (this->unk_40A & 0x40) {
-            this->unk_40A &= ~0x40;
+    Message_StartTextbox(play, this->textId, &this->actor);
+    if (this->flags & ENBABA_END_CONVERSATION) {
+        if (this->flags & ENBABA_GAVE_BLAST_MASK) {
+            this->flags &= ~ENBABA_GAVE_BLAST_MASK;
             func_80151BB4(play, 0x33);
         }
         func_80151BB4(play, 4);
     }
 }
 
-void func_80BA8C4C(PlayState* play, u16 nextEntrance) {
+void EnBaba_SetNextEntrance(PlayState* play, u16 nextEntrance) {
     play->nextEntranceIndex = nextEntrance;
     play->unk_1887F = 0x40;
     gSaveContext.nextTransition = 0x40;
     play->sceneLoadFlag = 0x14;
 }
```

Comment:
```suggestion
void EnBaba_SetNextEntrance(PlayState* play, u16 nextEntrance) {
    play->nextEntranceIndex = nextEntrance;
    play->transitionType = TRANS_TYPE_64;
    gSaveContext.nextTransitionType = TRANS_TYPE_64;
    play->transitionTrigger = TRANS_TRIGGER_START;
}
```
This is what this function looks like after my transition PR, so I don't think it's just setting the next entrance, it's also triggering the next transition. Maybe something simple like `EnBaba_TransitionScene` or `EnBaba_TriggerTransition`?

Reply:
Makes sense. I like `EnBaba_TriggerTransition` of the two you suggested.